<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Vignette 1: Getting Started with BANDLE • bandle</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Vignette 1: Getting Started with BANDLE">
<meta property="og:description" content="bandle">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">bandle</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.12</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/v01-getting-started.html">Vignette 1: Getting Started with BANDLE</a>
    </li>
    <li>
      <a href="../articles/v02-workflow.html">Vignette 2: A workflow for analysing differential localisation</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ococrook/bandle/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Vignette 1: Getting Started with BANDLE</h1>
                        <h4 data-toc-skip class="author">Oliver M. Crook</h4>
            <address class="author_afil">
      Department of Statistics, University of Oxford, UK<br><h4 data-toc-skip class="author">Lisa M. Breckels</h4>
            <address class="author_afil">
      Cambridge Centre for Proteomics, University of Cambridge, UK<br><small class="dont-index">Source: <a href="https://github.com/ococrook/bandle/blob/HEAD/vignettes/v01-getting-started.Rmd" class="external-link"><code>vignettes/v01-getting-started.Rmd</code></a></small>
      <div class="hidden name"><code>v01-getting-started.Rmd</code></div>

    </address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette provides an introduction to the BANDLE package <span class="citation">(Crook et al. 2021)</span> and follows a short theortical example of how to perform differential localisation analysis of quantitative proteomics data using the BANDLE model. Explanation and general recommendations of the input parameters are provided here. For a more comprehensive workflow which follows a real-life use case, please see the second vignette in this package.</p>
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># options(width = 150)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="http://github.com/ococrook/bandle" class="external-link">"bandle"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://ggplot2.tidyverse.org" class="external-link">"ggplot2"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lgatto/pRoloc" class="external-link">"pRoloc"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lgatto/pRolocdata" class="external-link">"pRolocdata"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://lgatto.github.io/MSnbase" class="external-link">"MSnbase"</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"pheatmap"</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://sjmgarnier.github.io/viridis/" class="external-link">"viridis"</a></span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">setStockcol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span>, <span class="st">"90"</span><span class="op">)</span><span class="op">)</span> <span class="co"># settransparent colours</span></code></pre></div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Bayesian ANalysis of Differential Localisation Experiments (BANDLE) is an integrative semi-supervised functional mixture model, developed by <span class="citation">(Crook et al. 2021)</span>, to obtain the probability of a protein being differentially localised between two conditions.</p>
<p>In this vignette we walk users through how to install and use the R <span class="citation">(R Development Core Team 2011)</span> Bioconductor <span class="citation">(Gentleman et al. 2004)</span> <a href="https://github.com/ococrook/bandle" class="external-link"><code>bandle</code> package</a> by simulating a well-defined differential localisation experiment from spatial proteomics data from the <code>pRolocdata</code> package <span class="citation">(Gatto et al. 2014)</span>.</p>
<p>The BANDLE method uses posterior Bayesian computations performed using Markov-chain Monte-Carlo (MCMC) and thus uncertainty estimates are available <span class="citation">(Gilks, Richardson, and Spiegelhalter 1995)</span>. Throughout this vignette we use the term <em>differentially localised</em> to pertain to proteins which are assigned to different sub-cellular localisations between two conditions. One of the main outputs of BANDLE is the probability that a protein is differentially localised between two conditions.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The package can be installed with the <code>BiocManager</code> package:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"bandle"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="the-data">The data<a class="anchor" aria-label="anchor" href="#the-data"></a>
</h2>
<p>In this vignette and <span class="citation">Crook et al. (2021)</span>, the main data source that we use to study differential protein sub-cellular localisation are data from high-throughput mass spectrometry-based experiments. The data from these types of experiments traditionally yield a matrix of measurements wherein we have, for example, PSMs, peptides or proteins along the rows, and samples/channels/fractions along the columns. The <code>bandle</code> package uses the <code>MSnSet</code> class as implemented in the Bioconductor <em><a href="https://bioconductor.org/packages/3.14/MSnbase" class="external-link">MSnbase</a></em> package and thus requires users to import and store their data as a <code>MSnSet</code> instance. For more details on how to create a <code>MSnSet</code> see the relevant vignettes in <em><a href="https://bioconductor.org/packages/3.14/pRoloc" class="external-link">pRoloc</a></em>. The <em><a href="https://bioconductor.org/packages/3.14/pRolocdata" class="external-link">pRolocdata</a></em> experiment data package is a good starting place to look for test data. This data package contains tens of quantitative proteomics experiments, stored as <code>MSnSet</code>s.</p>
<div class="section level3">
<h3 id="a-well-defined-theoretical-example">A well-defined theoretical example<a class="anchor" aria-label="anchor" href="#a-well-defined-theoretical-example"></a>
</h3>
<p>To get started with the basics of using <code>bandle</code> we begin by generating a simple example dataset which simulates a differential localisation experiment (please see the second vignette in this package for a full real-life biological use case). In this example data, the key elements are replicates, and a perturbation of interest. There is code within the <em><a href="https://bioconductor.org/packages/3.14/bandle" class="external-link">bandle</a></em> package to simulate an example experiment.</p>
<p>In the code chunk below we begin by loading the <em><a href="https://bioconductor.org/packages/3.14/pRolocdata" class="external-link">pRolocdata</a></em> package to obtain a spatial proteomics dataset. This will be the basis of our simulation which will use boostrapping to generate new datasets. The dataset we have chosen to load is a dataset from 2009 (<code>tan2009r1</code>). This is data from a early LOPIT experiment performed on Drosophila embryos <span class="citation">(Tan et al. 2009)</span>. The aim of this experiment was to apply LOPIT to an organism with heterogeneous cell types. This experiment used four isotopes across four distinct fractions and thus yielded four measurements (features) per protein profile. We visualise the data by using principal components analysis.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/lgatto/pRolocdata" class="external-link">"pRolocdata"</a></span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"tan2009r1"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/plot2D.html" class="external-link">plot2D</a></span><span class="op">(</span><span class="va">tan2009r1</span>,
       main <span class="op">=</span> <span class="st">"An example spatial proteomics datasets"</span>, 
       grid <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/addLegend.html" class="external-link">addLegend</a></span><span class="op">(</span><span class="va">tan2009r1</span>, where <span class="op">=</span> <span class="st">"topleft"</span>, cex <span class="op">=</span> <span class="fl">0.7</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/loadpkgdat-1.png" width="175"></p>
<p>The following code chuck simulates a differential localisation experiment. It will generate <code>numRep/2</code> of each a control and treatment condition. We will also simulate relocalisations for <code>numDyn</code> proteins.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">tansim</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle-sim.html">sim_dynamic</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">tan2009r1</span>,
                      numRep <span class="op">=</span> <span class="fl">6L</span>,
                      numDyn <span class="op">=</span> <span class="fl">100L</span><span class="op">)</span></code></pre></div>
<p>The list of the 6 simulated experiments are found in <code>tansim$lopitrep</code>. Each one is an <code>MSnSet</code> instance (the standard data container for proteomics experimental data). The first 3 are the simulated control experiments (see <code>tansim$lopitrep[1:3]</code>), and the following 3 in the list are the treatment condition simulated experiments (see <code>tansim$lopitrep[4:6]</code>). We can plot them using the <code>plot2D</code> function from <code>pRoloc</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">plot_title</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Replicate "</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, <span class="st">" condition"</span>, <span class="st">" A"</span><span class="op">)</span>, 
               <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Replicate "</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, <span class="st">" condition"</span>, <span class="st">" B"</span><span class="op">)</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">z</span><span class="op">)</span> 
    <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/plot2D.html" class="external-link">plot2D</a></span><span class="op">(</span><span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">[[</span><span class="va">z</span><span class="op">]</span><span class="op">]</span>, grid <span class="op">=</span> <span class="cn">FALSE</span>, main <span class="op">=</span> <span class="va">plot_title</span><span class="op">[</span><span class="va">z</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/plotsims-1.png" width="175"></p>
<p>For understanding, exploring and visualizing individual spatial proteomics experiments, see the vignettes in <code>pRoloc</code> and <code>MSnbase</code> packages.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">## MSnSet (storageMode: lockedEnvironment)</span>
<span class="co">## assayData: 888 features, 4 samples </span>
<span class="co">##   element names: exprs </span>
<span class="co">## protocolData: none</span>
<span class="co">## phenoData</span>
<span class="co">##   sampleNames: X114 X115 X116 X117</span>
<span class="co">##   varLabels: Fractions</span>
<span class="co">##   varMetadata: labelDescription</span>
<span class="co">## featureData</span>
<span class="co">##   featureNames: P20353 P53501 ... P07909 (888 total)</span>
<span class="co">##   fvarLabels: FBgn Protein.ID ... knn.scores (18 total)</span>
<span class="co">##   fvarMetadata: labelDescription</span>
<span class="co">## experimentData: use 'experimentData(object)'</span>
<span class="co">##   pubMedIds: 19317464 </span>
<span class="co">## Annotation:  </span>
<span class="co">## - - - Processing information - - -</span>
<span class="co">## Added markers from  'mrk' marker vector. Thu Jul 16 22:53:44 2015 </span>
<span class="co">## Performed knn prediction (k=10) Tue Apr 26 10:06:20 2022 </span>
<span class="co">##  MSnbase version: 1.17.12</span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="preparing-for-bandle-analysis">Preparing for <code>bandle</code> analysis<a class="anchor" aria-label="anchor" href="#preparing-for-bandle-analysis"></a>
</h2>
<p>The main function of the package is <code>bandle</code>, this uses a complex model to analyse the data. Markov-Chain Monte-Carlo (MCMC) is used to sample the posterior distribution of parameters and latent variables. From which statistics of interest can be computed. Here we only run a few iterations for brevity but typically one needs to run thousands of iterations to ensure convergence, as well as multiple parallel chains.</p>
<div class="section level3">
<h3 id="fitting-gaussian-processes">Fitting Gaussian processes<a class="anchor" aria-label="anchor" href="#fitting-gaussian-processes"></a>
</h3>
<p>First, we need to fit non-parametric regression functions to the markers profiles, upon which we place our analysis. This uses Gaussian processes. The <code>fitGPmaternPC</code> function can be used and fits some default penalised complexity priors (see <code><a href="../reference/bandle-gpfit.html">?fitGP</a></code>), which works well. However, these can be altered, which is demonstrated in the next code chunk</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="va">gpParams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="../reference/bandle-gpfit.html">fitGPmaternPC</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>We apply the <code>fitGPmaternPC</code> function to each datasets by calling <code>lapply</code> over the <code>tansim$lopitrep</code> list of datasets. The output of <code>fitGPmaternPC</code> returns a list of posterior predictive means and standard deviations. As well as MAP hyperparamters for the GP.</p>
<p>The <code>plotGPmatern</code> function can be used to plot the profiles for each class in each replicate condition with the posterior predictive distributions are overlayed with markers protein profiles.</p>
<p>For example, to plot the predictive distributions of the first dataset,</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="../reference/bandle-gpfit.html">plotGPmatern</a></span><span class="op">(</span><span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, params <span class="op">=</span> <span class="va">gpParams</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/plotgps-1.png" width="200"></p>
<p>The prior needs to form a <code>K*3</code> matrix. <code>K</code> corresponds to the number of subcellular classes in the data, and 3 columns for (1) the prior, (2) length-scale amplitude and (3) standard deviation parameters (see <code>hyppar</code> in <code><a href="../reference/bandle-gpfit.html">?fitGP</a></code>). Increasing these values, increases the shrinkage. For more details see the manuscript by <span class="citation">Crook et al. (2021)</span>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">K</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/pSet-class.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getMarkerClasses.html" class="external-link">getMarkerClasses</a></span><span class="op">(</span><span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, fcol <span class="op">=</span> <span class="st">"markers"</span><span class="op">)</span><span class="op">)</span>
<span class="va">pc_prior</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="cn">NA</span>, ncol <span class="op">=</span> <span class="fl">3</span>, <span class="va">K</span><span class="op">)</span>
<span class="va">pc_prior</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq.int</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="va">K</span><span class="op">)</span>, <span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">10</span>, <span class="fl">60</span>, <span class="fl">250</span><span class="op">)</span>,
                                       each <span class="op">=</span> <span class="va">K</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></code></pre></div>
<p>Now we have generated these complexity priors we can pass them as an argument to the <code>fitGPmaternPC</code> function. For example,</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gpParams</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span>,
                   <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="../reference/bandle-gpfit.html">fitGPmaternPC</a></span><span class="op">(</span><span class="va">x</span>, hyppar <span class="op">=</span> <span class="va">pc_prior</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p>By looking at the plot of posterior predictives using the <code>gpParams</code> we can see the GP fit looks sensible.</p>
</div>
<div class="section level3">
<h3 id="setting-the-prior-on-the-weights">Setting the prior on the weights<a class="anchor" aria-label="anchor" href="#setting-the-prior-on-the-weights"></a>
</h3>
<p>The next step is to set up the matrix Dirichlet prior on the mixing weights. These weights are defined across datasets so these are slightly different to mixture weights in usual mixture models. The <span class="math inline">\((i,j)^{th}\)</span> entry is the prior probability that a protein localises to organelle <span class="math inline">\(i\)</span> in the control and <span class="math inline">\(j\)</span> in the treatment. This mean that off-diagonal terms have a different interpretation to diagonal terms. Since we expect re-localisation to be rare, off-diagonal terms should be small. The following functions help set up the priors and how to interpret them. The parameter <code>q</code> allow us to check the prior probability that more than <code>q</code> differential localisations are expected.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">dirPrior</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/diag.html" class="external-link">diag</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="va">K</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="fl">0.001</span>, nrow <span class="op">=</span> <span class="va">K</span>, ncol <span class="op">=</span> <span class="va">K</span><span class="op">)</span>
<span class="va">predDirPrior</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle-prior.html">prior_pred_dir</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>,
                               dirPrior <span class="op">=</span> <span class="va">dirPrior</span>,
                               q <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></code></pre></div>
<p>The mean number of re-localisations is small:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">predDirPrior</span><span class="op">$</span><span class="va">meannotAlloc</span></code></pre></div>
<pre><code><span class="co">## [1] 0.3515246</span></code></pre>
<p>The prior probability that more than <code>q</code> differential localisations are expected is small</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">predDirPrior</span><span class="op">$</span><span class="va">tailnotAlloc</span></code></pre></div>
<pre><code><span class="co">## [1] 6e-04</span></code></pre>
<p>The full prior predictive can be visualised as histogram. The prior probability that proteins are allocated to different components between datasets concentrates around 0.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">predDirPrior</span><span class="op">$</span><span class="va">priornotAlloc</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/unnamed-chunk-5-1.png" width="175"></p>
<p>For most use-cases we indeed expect the number of differential localisations to be small. However, there may be specific cases where one may expect the either a smaller or larger number of differential localisations. Users could try testing different values for the <code>dirPrior</code> for example, replacing 0.001 with 0.0005 or smaller, for larger datasets to bring the number of expected re-localisations inline with the biological expectation, and visa-versa when we expect the number of proteins to have changed to be higher.</p>
</div>
</div>
<div class="section level2">
<h2 id="running-the-bandle-function">Running the <code>bandle</code> function<a class="anchor" aria-label="anchor" href="#running-the-bandle-function"></a>
</h2>
<p>We are now ready to run the main <code>bandle</code> function. Remember to carefully select the datasets and replicates that define the control and treatment. As a reminder, in this introductory vignette we have used a small dataset and generated theoretical triplicates of each theoretical condition. Please see the second vignette in this package for a more detailed workflow and real biological use-case. In the below code chunk we run <code>bandle</code> for only 50 iterations for the convenience of building the vignette, but typically we’d recommend you run the number of iterations (<code>numIter</code>) in the 1000s.</p>
<p>Remember: the first 3 datasets are the first 3 elements of <code>tansim</code> and the final 3 elements are the “treatment” triplicate datasets.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">control</span> <span class="op">&lt;-</span> <span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> 
<span class="va">treatment</span> <span class="op">&lt;-</span> <span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span>

<span class="va">bandleres</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle.html">bandle</a></span><span class="op">(</span>objectCond1 <span class="op">=</span> <span class="va">control</span>,
                    objectCond2 <span class="op">=</span> <span class="va">treatment</span>,
                    numIter <span class="op">=</span> <span class="fl">20</span>,  <span class="co"># usually 10,000</span>
                    burnin <span class="op">=</span> <span class="fl">5L</span>,   <span class="co"># usually 5,000</span>
                    thin <span class="op">=</span> <span class="fl">1L</span>,     <span class="co"># usually 20</span>
                    gpParams <span class="op">=</span> <span class="va">gpParams</span>,
                    pcPrior <span class="op">=</span> <span class="va">pc_prior</span>,
                    numChains <span class="op">=</span> <span class="fl">1</span>,  <span class="co"># usually &gt;=4</span>
                    dirPrior <span class="op">=</span> <span class="va">dirPrior</span><span class="op">)</span></code></pre></div>
<p>The <code>bandle</code> function generates an object of class <code>bandleParams</code>. The <code>show</code> method indicates the number of parallel chains that were run, this should typically be greater than 4 (here we use 1 just as a demo). Normally, we should also assess convergence but this omitted for the moment so that we can move forward with the analysis. Please see the end of the vignette for convergence plots.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandleres</span></code></pre></div>
<pre><code><span class="co">## Object of class "bandleParams"</span>
<span class="co">## Method: bandle </span>
<span class="co">## Number of chains: 1</span></code></pre>
</div>
<div class="section level2">
<h2 id="analysing-bandle-output">Analysing <code>bandle</code> output<a class="anchor" aria-label="anchor" href="#analysing-bandle-output"></a>
</h2>
<p>Before we can begin to extract protein allocation information and a list of proteins which are differentially localised between conditions, we first need to populate the <code>bandleres</code> object by calling the <code>bandleProcess</code> function.</p>
<div class="section level3">
<h3 id="populating-a-bandleres-object">Populating a <code>bandleres</code> object<a class="anchor" aria-label="anchor" href="#populating-a-bandleres-object"></a>
</h3>
<p>Currently, the summary slots of the <code>bandleres</code> object are empty. The <code>summaries</code> function accesses them.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bandleParams.html">summaries</a></span><span class="op">(</span><span class="va">bandleres</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [[1]]</span>
<span class="co">## An object of class "bandleSummary"</span>
<span class="co">## Slot "posteriorEstimates":</span>
<span class="co">## DataFrame with 0 rows and 0 columns</span>
<span class="co">## </span>
<span class="co">## Slot "diagnostics":</span>
<span class="co">## &lt;0 x 0 matrix&gt;</span>
<span class="co">## </span>
<span class="co">## Slot "bandle.joint":</span>
<span class="co">## &lt;0 x 0 matrix&gt;</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## [[2]]</span>
<span class="co">## An object of class "bandleSummary"</span>
<span class="co">## Slot "posteriorEstimates":</span>
<span class="co">## DataFrame with 0 rows and 0 columns</span>
<span class="co">## </span>
<span class="co">## Slot "diagnostics":</span>
<span class="co">## &lt;0 x 0 matrix&gt;</span>
<span class="co">## </span>
<span class="co">## Slot "bandle.joint":</span>
<span class="co">## &lt;0 x 0 matrix&gt;</span></code></pre>
<p>These can be populated as follows</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bandleres</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle-process.html">bandleProcess</a></span><span class="op">(</span><span class="va">bandleres</span><span class="op">)</span></code></pre></div>
<p>These slots have now been populated</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="fu"><a href="../reference/bandleParams.html">summaries</a></span><span class="op">(</span><span class="va">bandleres</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##      Length Class         Mode</span>
<span class="co">## [1,] 1      bandleSummary S4  </span>
<span class="co">## [2,] 1      bandleSummary S4</span></code></pre>
<div class="section level4">
<h4 id="bandle-results">
<code>bandle</code> results<a class="anchor" aria-label="anchor" href="#bandle-results"></a>
</h4>
<p>We can save the results by calling <code>summaries</code>. We see that it is of length 2. 1 for control and 1 for treatment.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandleParams.html">summaries</a></span><span class="op">(</span><span class="va">bandleres</span><span class="op">)</span>
<span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/pSet-class.html" class="external-link">length</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 2</span></code></pre>
<p>There are a number of slots,</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">res</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Formal class 'bandleSummary' [package "bandle"] with 3 slots</span>
<span class="co">##   ..@ posteriorEstimates:Formal class 'DFrame' [package "S4Vectors"] with 6 slots</span>
<span class="co">##   .. .. ..@ rownames       : chr [1:677] "P20353" "P53501" "Q7KU78" "P04412" ...</span>
<span class="co">##   .. .. ..@ nrows          : int 677</span>
<span class="co">##   .. .. ..@ listData       :List of 7</span>
<span class="co">##   .. .. .. ..$ bandle.allocation               : Named chr [1:677] "PM" "Golgi" "Ribosome 40S" "PM" ...</span>
<span class="co">##   .. .. .. .. ..- attr(*, "names")= chr [1:677] "P20353" "P53501" "Q7KU78" "P04412" ...</span>
<span class="co">##   .. .. .. ..$ bandle.probability              : Named num [1:677] 1 0.995 1 0.953 0.997 ...</span>
<span class="co">##   .. .. .. .. ..- attr(*, "names")= chr [1:677] "P20353" "P53501" "Q7KU78" "P04412" ...</span>
<span class="co">##   .. .. .. ..$ bandle.outlier                  : num [1:677] 0 0 0 0 0 0 0 0 0 0 ...</span>
<span class="co">##   .. .. .. ..$ bandle.probability.lowerquantile: Named num [1:677] 1 0.983 1 0.871 0.991 ...</span>
<span class="co">##   .. .. .. .. ..- attr(*, "names")= chr [1:677] "P20353" "P53501" "Q7KU78" "P04412" ...</span>
<span class="co">##   .. .. .. ..$ bandle.probability.upperquantile: Named num [1:677] 1 0.999 1 0.997 0.999 ...</span>
<span class="co">##   .. .. .. .. ..- attr(*, "names")= chr [1:677] "P20353" "P53501" "Q7KU78" "P04412" ...</span>
<span class="co">##   .. .. .. ..$ bandle.mean.shannon             : Named num [1:677] 3.05e-06 1.42e-03 0.00 1.92e-02 0.00 ...</span>
<span class="co">##   .. .. .. .. ..- attr(*, "names")= chr [1:677] "P20353" "P53501" "Q7KU78" "P04412" ...</span>
<span class="co">##   .. .. .. ..$ bandle.differential.localisation: Named num [1:677] 0 0 0 0 0 0 0 0 0 0 ...</span>
<span class="co">##   .. .. .. .. ..- attr(*, "names")= chr [1:677] "P20353" "P53501" "Q7KU78" "P04412" ...</span>
<span class="co">##   .. .. ..@ elementType    : chr "ANY"</span>
<span class="co">##   .. .. ..@ elementMetadata: NULL</span>
<span class="co">##   .. .. ..@ metadata       : list()</span>
<span class="co">##   ..@ diagnostics       : logi [1, 1] NA</span>
<span class="co">##   ..@ bandle.joint      : num [1:677, 1:11] 1.03e-119 9.15e-137 3.80e-31 1.63e-147 1.00e-68 ...</span>
<span class="co">##   .. ..- attr(*, "dimnames")=List of 2</span>
<span class="co">##   .. .. ..$ : chr [1:677] "P20353" "P53501" "Q7KU78" "P04412" ...</span>
<span class="co">##   .. .. ..$ : chr [1:11] "Cytoskeleton" "ER" "Golgi" "Lysosome" ...</span></code></pre>
<p>The main one of interest is the <code>posteriorEstimates</code> slot,</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bandleParams.html">posteriorEstimates</a></span><span class="op">(</span><span class="va">res</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DataFrame with 677 rows and 7 columns</span>
<span class="co">##        bandle.allocation bandle.probability bandle.outlier</span>
<span class="co">##              &lt;character&gt;          &lt;numeric&gt;      &lt;numeric&gt;</span>
<span class="co">## P20353                PM           0.999997              0</span>
<span class="co">## P53501             Golgi           0.994623              0</span>
<span class="co">## Q7KU78      Ribosome 40S           1.000000              0</span>
<span class="co">## P04412                PM           0.953286              0</span>
<span class="co">## Q7KJ73      Ribosome 60S           0.997251              0</span>
<span class="co">## ...                  ...                ...            ...</span>
<span class="co">## Q95TL8                ER           1.000000              0</span>
<span class="co">## P25007        Proteasome           0.999999              0</span>
<span class="co">## P41374      Cytoskeleton           0.999933              0</span>
<span class="co">## Q8SZM1        Peroxisome           0.939575              0</span>
<span class="co">## P07909           Nucleus           1.000000              0</span>
<span class="co">##        bandle.probability.lowerquantile bandle.probability.upperquantile</span>
<span class="co">##                               &lt;numeric&gt;                        &lt;numeric&gt;</span>
<span class="co">## P20353                         0.999993                         1.000000</span>
<span class="co">## P53501                         0.982557                         0.999237</span>
<span class="co">## Q7KU78                         1.000000                         1.000000</span>
<span class="co">## P04412                         0.870518                         0.996670</span>
<span class="co">## Q7KJ73                         0.991004                         0.999203</span>
<span class="co">## ...                                 ...                              ...</span>
<span class="co">## Q95TL8                         1.000000                         1.000000</span>
<span class="co">## P25007                         0.999997                         1.000000</span>
<span class="co">## P41374                         0.999584                         0.999998</span>
<span class="co">## Q8SZM1                         0.772711                         0.987770</span>
<span class="co">## P07909                         1.000000                         1.000000</span>
<span class="co">##        bandle.mean.shannon bandle.differential.localisation</span>
<span class="co">##                  &lt;numeric&gt;                        &lt;numeric&gt;</span>
<span class="co">## P20353         3.05311e-06                                0</span>
<span class="co">## P53501         1.41648e-03                                0</span>
<span class="co">## Q7KU78         0.00000e+00                                0</span>
<span class="co">## P04412         1.92401e-02                                0</span>
<span class="co">## Q7KJ73         0.00000e+00                                0</span>
<span class="co">## ...                    ...                              ...</span>
<span class="co">## Q95TL8         0.00000e+00                                0</span>
<span class="co">## P25007         0.00000e+00                                0</span>
<span class="co">## P41374         0.00000e+00                                0</span>
<span class="co">## Q8SZM1         4.80859e-03                                1</span>
<span class="co">## P07909         1.03218e-07                                0</span></code></pre>
<p>This output object is a <code>data.frame</code> containing the protein allocations and associated localisation probabilities (including the upper and lower quantiles of the allocation probability distribution), the mean Shannon entropy and the <code>bandle.differential.localisation</code> probability.</p>
</div>
</div>
<div class="section level3">
<h3 id="extracting-posteriors-and-allocation-results">Extracting posteriors and allocation results<a class="anchor" aria-label="anchor" href="#extracting-posteriors-and-allocation-results"></a>
</h3>
<p>We create two new objects <code>pe1</code> and <code>pe2</code> in the below code chunk which contain the output of the <code>posteriorEstimates</code> slot.</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pe1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandleParams.html">posteriorEstimates</a></span><span class="op">(</span><span class="va">res</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="va">pe2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandleParams.html">posteriorEstimates</a></span><span class="op">(</span><span class="va">res</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p>One quantity of interest is the protein allocations, which we can plot as a barplot.</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">alloc1</span> <span class="op">&lt;-</span> <span class="va">pe1</span><span class="op">$</span><span class="va">bandle.allocation</span>
<span class="va">alloc2</span> <span class="op">&lt;-</span> <span class="va">pe2</span><span class="op">$</span><span class="va">bandle.allocation</span>

<span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">2</span><span class="op">)</span>, oma <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">6</span>,<span class="fl">2</span>,<span class="fl">2</span>,<span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">alloc1</span><span class="op">)</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,
        las <span class="op">=</span> <span class="fl">2</span>, main <span class="op">=</span> <span class="st">"Protein allocation: control"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/barplot.html" class="external-link">barplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">alloc2</span><span class="op">)</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">2</span><span class="op">]</span>,
        las <span class="op">=</span> <span class="fl">2</span>, main <span class="op">=</span> <span class="st">"Protein allocation: treatment"</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/barplotalloc-1.png" width="200"></p>
<p>The barplot tells us for this example that <code>bandle</code> has allocated the majority of unlabelled proteins to the ER, followed by the Golgi (irrespective of the posterior probability).</p>
<p>The associated posterior estimates are located in the <code>bandle.probability</code> column.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pe_alloc1</span> <span class="op">&lt;-</span> <span class="va">pe1</span><span class="op">$</span><span class="va">bandle.probability</span>
<span class="va">pe_alloc2</span> <span class="op">&lt;-</span> <span class="va">pe1</span><span class="op">$</span><span class="va">bandle.probability</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="allocation-probabilities">Allocation probabilities<a class="anchor" aria-label="anchor" href="#allocation-probabilities"></a>
</h3>
<p>The full allocation probabilities are stored in the <code>tagm.joint</code> slot. These can be visualised in a heatmap</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bjoint_control</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandleParams.html">bandleJoint</a></span><span class="op">(</span><span class="fu"><a href="../reference/bandleParams.html">summaries</a></span><span class="op">(</span><span class="va">bandleres</span><span class="op">)</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">bjoint_control</span>, cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>, color <span class="op">=</span> <span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">25</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/heatmap_control-1.png" width="175"></p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bjoint_treatment</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandleParams.html">bandleJoint</a></span><span class="op">(</span><span class="fu"><a href="../reference/bandleParams.html">summaries</a></span><span class="op">(</span><span class="va">bandleres</span><span class="op">)</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">bjoint_treatment</span>, cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>, color <span class="op">=</span> <span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">viridis</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">25</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/heatmap_treatment-1.png" width="175"></p>
</div>
<div class="section level3">
<h3 id="predicting-the-subcellular-location">Predicting the subcellular location<a class="anchor" aria-label="anchor" href="#predicting-the-subcellular-location"></a>
</h3>
<p>We can append the results to our original <code>MSnSet</code> datasets using the <code>bandlePredict</code> function.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle-predict.html">bandlePredict</a></span><span class="op">(</span><span class="va">control</span>, 
                    <span class="va">treatment</span>, 
                    params <span class="op">=</span> <span class="va">bandleres</span>, 
                    fcol <span class="op">=</span> <span class="st">"markers"</span><span class="op">)</span>
<span class="va">res_control</span> <span class="op">&lt;-</span> <span class="va">xx</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">res_treatment</span> <span class="op">&lt;-</span> <span class="va">xx</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p>The output is a <code>list</code> of <code>MSnSets</code>. In this example, we have 3 for the control and 3 for the treatment.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/pSet-class.html" class="external-link">length</a></span><span class="op">(</span><span class="va">res_control</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 3</span></code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/pSet-class.html" class="external-link">length</a></span><span class="op">(</span><span class="va">res_treatment</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 3</span></code></pre>
<p>The results are appended to the <strong>first</strong> <code>MSnSet</code> feature data slot for each condition.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fvarLabels</a></span><span class="op">(</span><span class="va">res_control</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] "FBgn"                             "Protein.ID"                      </span>
<span class="co">##  [3] "Flybase.Symbol"                   "AccessionNo"                     </span>
<span class="co">##  [5] "EntryName"                        "AccessionNoAll"                  </span>
<span class="co">##  [7] "EntryNameAll"                     "No.peptide.IDs"                  </span>
<span class="co">##  [9] "Mascot.score"                     "No.peptide.quantified"           </span>
<span class="co">## [11] "PLSDA"                            "pd.2013"                         </span>
<span class="co">## [13] "pd.markers"                       "markers.orig"                    </span>
<span class="co">## [15] "markers"                          "markers.tl"                      </span>
<span class="co">## [17] "knn"                              "knn.scores"                      </span>
<span class="co">## [19] "bandle.allocation"                "bandle.probability"              </span>
<span class="co">## [21] "bandle.probability.lowerquantile" "bandle.probability.upperquantile"</span>
<span class="co">## [23] "bandle.mean.shannon"              "bandle.differential.localisation"</span>
<span class="co">## [25] "bandle.outlier"                   "bandle.joint"</span></code></pre>
<p>To access them use the <code>fData</code> function</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">res_control</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">bandle.probability</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">res_control</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">bandle.allocation</span></code></pre></div>
<div class="section level4">
<h4 id="thresholding-on-protein-allocations">Thresholding on protein allocations<a class="anchor" aria-label="anchor" href="#thresholding-on-protein-allocations"></a>
</h4>
<p>It is common practice in supervised machine learning to set a specific threshold on which to define new assignments/allocations, below which classifications are left unassigned/unknown. Indeed, we do not expect the whole subcellular diversity to be represented by the 11 niches defined here, we expect there to be many more, many of which will be multiply localised within the cell. It is important to allow for the possibility of proteins to reside in multiple locations (this information is available in the <code>bandle.joint</code> slot - see above for more details on multiple location).</p>
<p>As we are using a Bayesian model the outputs of the classifier are probabilities. This not only allows us to look at the distribution of probabilities over all subcellular classes but also allows us to extract a probability threshold on which we can define new assignments.</p>
<p>The subcellular allocations are located in the <code>bandle.allocation</code> column of the <code>fData</code> slot and the posteriors are located in the <code>bandle.probability</code> slot. We can use the <code>getPredictions</code> function from the <code>pRoloc</code> package to return a set of predicted localisations according to if they meet a probability threshold.</p>
<p>For example, in the below code chunk we set a 1% FDR for assigning proteins a subcellular nice, below which we leave them unlabelled.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_control</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getPredictions.html" class="external-link">getPredictions</a></span><span class="op">(</span><span class="va">res_control</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, 
                                   fcol <span class="op">=</span> <span class="st">"bandle.allocation"</span>,                   
                                   scol <span class="op">=</span> <span class="st">"bandle.probability"</span>,                   
                                   mcol <span class="op">=</span> <span class="st">"markers"</span>,                   
                                   t <span class="op">=</span> <span class="fl">.99</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ans</span>
<span class="co">##  Cytoskeleton            ER         Golgi      Lysosome mitochondrion </span>
<span class="co">##            13           216           128            10            84 </span>
<span class="co">##       Nucleus    Peroxisome            PM    Proteasome  Ribosome 40S </span>
<span class="co">##            40             7           136            45            76 </span>
<span class="co">##  Ribosome 60S       unknown </span>
<span class="co">##            51            82</span></code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_treatment</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getPredictions.html" class="external-link">getPredictions</a></span><span class="op">(</span><span class="va">res_treatment</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, 
                                   fcol <span class="op">=</span> <span class="st">"bandle.allocation"</span>,                   
                                   scol <span class="op">=</span> <span class="st">"bandle.probability"</span>,                   
                                   mcol <span class="op">=</span> <span class="st">"markers"</span>,                   
                                   t <span class="op">=</span> <span class="fl">.99</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ans</span>
<span class="co">##  Cytoskeleton            ER         Golgi      Lysosome mitochondrion </span>
<span class="co">##            17           191           120            16            83 </span>
<span class="co">##       Nucleus    Peroxisome            PM    Proteasome  Ribosome 40S </span>
<span class="co">##            48            14           134            50            71 </span>
<span class="co">##  Ribosome 60S       unknown </span>
<span class="co">##            52            92</span></code></pre>
<p>We may also wish to take into account the probability of the protein being an outlier and thus use the results in the <code>bandle.outlier</code> column of the feature data. We could calculate the product of the posterior and the outlier (as they are both probabilities) to obtain a localisation score that takes into account the outlier model. More details on this are found in the second vignette of this package.</p>
</div>
</div>
<div class="section level3">
<h3 id="differential-localisation-probability">Differential localisation probability<a class="anchor" aria-label="anchor" href="#differential-localisation-probability"></a>
</h3>
<p>As previously mentioned the term “differentially localised” is used to pertain to proteins which are assigned to different sub-cellular localisations between two conditions. For the majority of users this is the main output they are keen to extract using the BANDLE method.</p>
<p>Following on from the above example, after extracting posterior estimates for each condition using the <code>summaries</code> function we can also access the differential localisation probability as it is stored in the <code>bandle.differential.localisation</code> column of the <code>data.frames</code> of <code>pe1</code> and <code>pe2</code>, in the above sections.</p>
<p>The differential localisation probability tells us which proteins are most likely to <em>differentially localise</em>. We can for example, take a 1% FDR and examine how many proteins get a differential probability greater than 0.99.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">diffloc_probs</span> <span class="op">&lt;-</span> <span class="va">pe1</span><span class="op">$</span><span class="va">bandle.differential.localisation</span>
<span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">head</a></span><span class="op">(</span><span class="va">diffloc_probs</span>, <span class="fl">50</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##     P20353     P53501     Q7KU78     P04412     Q7KJ73     Q9VM65     Q9VCK0 </span>
<span class="co">## 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 </span>
<span class="co">##     B7Z0W3     Q9V415     Q00174     Q9V769     Q27593     Q9V780     P19109 </span>
<span class="co">## 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 </span>
<span class="co">##     Q95NR4     O77047     Q8SXD0     Q9VBV5     Q9VBU5     Q8IA62     Q9Y105 </span>
<span class="co">## 0.00000000 0.06666667 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 </span>
<span class="co">##     Q9W2M4     P26308     Q9VP77     Q5U0Z2     Q960E2     Q9VLJ6     Q9VJ39 </span>
<span class="co">## 0.00000000 1.00000000 1.00000000 0.00000000 0.00000000 1.00000000 1.00000000 </span>
<span class="co">##     Q9VTX8     Q9VTZ5     B7Z0C1     Q9VRJ4     M9PCB7     P46150     A1ZBH5 </span>
<span class="co">## 0.00000000 0.53333333 0.00000000 0.00000000 0.00000000 0.33333333 0.00000000 </span>
<span class="co">##     Q9W3M7     A8DZ29     Q9VN14     Q9VZL3     M9PC99     Q86BP3     Q9W3N9 </span>
<span class="co">## 0.00000000 0.66666667 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 </span>
<span class="co">##     Q7K5M6     P16620     P48375     Q9VMB9     Q9VI55     Q9VU58     Q9V498 </span>
<span class="co">## 0.00000000 0.00000000 0.00000000 0.00000000 0.00000000 0.06666667 0.00000000 </span>
<span class="co">##     Q9V496 </span>
<span class="co">## 1.00000000</span></code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/pSet-class.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">diffloc_probs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">diffloc_probs</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span> <span class="op">&gt;</span> <span class="fl">0.99</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 82</span></code></pre>
<p>We find there are 82 proteins above this threshold.</p>
<p>This can also be seen on a rank plot</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://lgatto.github.io/MSnbase/reference/plot-methods.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">diffloc_probs</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">diffloc_probs</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">]</span>,
     col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">3</span><span class="op">]</span>, pch <span class="op">=</span> <span class="fl">19</span>, ylab <span class="op">=</span> <span class="st">"Probability"</span>,
     xlab <span class="op">=</span> <span class="st">"Rank"</span>, main <span class="op">=</span> <span class="st">"Differential localisation rank plot"</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/extractdiffloc-1.png" width="175"></p>
<p>In-line with our expectations, the rank plot indicates that most proteins are not differentially localised.</p>
<div class="section level4">
<h4 id="estimating-uncertainty">Estimating uncertainty<a class="anchor" aria-label="anchor" href="#estimating-uncertainty"></a>
</h4>
<div class="section level5">
<h5 id="applying-the-bootstrapdifflocprob-function">Applying the <code>bootstrapdiffLocprob</code> function<a class="anchor" aria-label="anchor" href="#applying-the-bootstrapdifflocprob-function"></a>
</h5>
<p>We can examine the top <code>n</code> proteins (here we use an example of <code>top = 100</code>) and produce bootstrap estimates of the uncertainty (note here the uncertainty is likely to be underestimated as we did not produce many MCMC samples). These can be visualised as ranked boxplots.</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span>
<span class="va">boot_t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle-differentiallocalisation.html">bootstrapdiffLocprob</a></span><span class="op">(</span>params <span class="op">=</span> <span class="va">bandleres</span>, top <span class="op">=</span> <span class="fl">100</span>,
                               Bootsample <span class="op">=</span> <span class="fl">5000</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">boot_t</span><span class="op">)</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span>,
        las <span class="op">=</span> <span class="fl">2</span>, ylab <span class="op">=</span> <span class="st">"Probability"</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>,
        main <span class="op">=</span> <span class="st">"Differential localisation \nprobability plot (top 100 proteins)"</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/diffloc_boot-1.png" width="175"></p>
</div>
<div class="section level5">
<h5 id="applying-the-binomdiffloc-function">Applying the <code>binomDiffLoc</code> function<a class="anchor" aria-label="anchor" href="#applying-the-binomdiffloc-function"></a>
</h5>
<p>Instead of applying the <code>bootstrapdiffLocprob</code> we could use the <code>binomDiffLoc</code> function to obtain credible intervals from the binomial distribution.</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bin_t</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle-differentiallocalisation.html">binomialDiffLocProb</a></span><span class="op">(</span>params <span class="op">=</span> <span class="va">bandleres</span>, top <span class="op">=</span> <span class="fl">100</span>,
                             nsample <span class="op">=</span> <span class="fl">5000</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/boxplot.html" class="external-link">boxplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">bin_t</span><span class="op">)</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span>,
        las <span class="op">=</span> <span class="fl">2</span>, ylab <span class="op">=</span> <span class="st">"Probability"</span>, ylim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>,
        main <span class="op">=</span> <span class="st">"Differential localisation \nprobability plot (top 100 proteins)"</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/diffloc_binom-1.png" width="175"></p>
</div>
<div class="section level5">
<h5 id="obtaining-probability-estimates">Obtaining probability estimates<a class="anchor" aria-label="anchor" href="#obtaining-probability-estimates"></a>
</h5>
<p>There are many ways we could obtain probability estimates from either of the above methods. We could, for example, take the mean of each protein estimate, or compute the cumulative error (there is not really a false discovery rate in Bayesian statistics) or we could threshold on the interval to reduce the number of differential localisations if you feel the model has been overconfident.</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># more robust estimate of probabilities</span>
<span class="va">dprobs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowMeans</a></span><span class="op">(</span><span class="va">bin_t</span><span class="op">)</span>

<span class="co"># compute cumulative error, there is not really a false discovery rate in</span>
<span class="co"># Bayesian statistics but you can look at the cumulative error rate</span>
<span class="va">ce</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cumsum.html" class="external-link">cumsum</a></span><span class="op">(</span><span class="fl">1</span>  <span class="op">-</span> <span class="va">dprobs</span><span class="op">)</span>

<span class="co"># you could threshold on the interval and this will reduce the number of</span>
<span class="co"># differential localisations</span>
<span class="va">qt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">bin_t</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/stats/quantile.html" class="external-link">quantile</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">.025</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level5">
<h5 id="the-expected-false-discovery-rate">The expected false discovery rate<a class="anchor" aria-label="anchor" href="#the-expected-false-discovery-rate"></a>
</h5>
<p>Instead of estimating the false discovery rate we can estimate the expected false discovery rate from the posterior probabilities at a particular threshold. This mean that for fixed threshold, we compute the expected proportion of false discoveries. Here is an example below. We can see that setting a probability threshold of 0.9 leads to an expected false discovery rate of less than <span class="math inline">\(0.5\%\)</span></p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bandle-EFDR.html">EFDR</a></span><span class="op">(</span><span class="va">diffloc_probs</span>, threshold <span class="op">=</span> <span class="fl">0.90</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 0.003100775</span></code></pre>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="visualising-differential-localisation">Visualising differential localisation<a class="anchor" aria-label="anchor" href="#visualising-differential-localisation"></a>
</h2>
<p>We can visualise the changes in localisation between conditions on an alluvial plot using the <code>plotTranslocations</code> function</p>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bandle-plots-translocations.html">plotTranslocations</a></span><span class="op">(</span><span class="va">bandleres</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/alluvial-1.png" width="175"></p>
<p>Or alternatively, on a chord (circos) diagram</p>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/bandle-plots-translocations.html">plotTranslocations</a></span><span class="op">(</span><span class="va">bandleres</span>, type <span class="op">=</span> <span class="st">"chord"</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/chord-1.png" width="175"></p>
<div class="section level4">
<h4 id="important-consideration">Important consideration<a class="anchor" aria-label="anchor" href="#important-consideration"></a>
</h4>
<p>These visualisations are showing the change in class label between the two conditions (as assigned by <code>bandle</code> i.e. the result stored in <code>bandle.allocation</code>). The results are taken directly from <code>bandleres</code> and thus no thresholding on the class label and posterior to allow for proteins to be left “unassigned” or unknown, is conducted. Furthermore, there is no thresholding on the <code>bandle.differential.localisation</code> probability.</p>
<p>It would be better to re-plot these figures with some thresholds on the above quantities, to get a better representation of what is moving. The easiest way to do this is to pass the <code>MSnSets</code> output after performing <code>bandlePredict</code> and <code>getPredictions</code>.</p>
<p>For example, first let’s identify which proteins get a high differential localisation probability,</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## identify which proteins get a high differential localisation probability</span>
<span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/featureData.html" class="external-link">fData</a></span><span class="op">(</span><span class="va">res_control</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">$</span><span class="va">bandle.differential.localisation</span> <span class="op">&gt;</span> <span class="fl">0.99</span><span class="op">)</span>

<span class="co">## create two new MSnSets with only these proteins</span>
<span class="va">res_dl_control</span> <span class="op">&lt;-</span> <span class="va">res_control</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">ind</span>, <span class="op">]</span>
<span class="va">res_dl_treatment</span> <span class="op">&lt;-</span> <span class="va">res_treatment</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="va">ind</span>, <span class="op">]</span></code></pre></div>
<p>Now we can plot only these 82 proteins which are deemed to move/differentially localise. We also specify where the prediction results are located e.g. <code>fcol = "bandle.allocation.pred"</code>.</p>
<div class="sourceCode" id="cb62"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## specify colour palette</span>
<span class="va">mycols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getStockcol.html" class="external-link">getStockcol</a></span><span class="op">(</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getMarkerClasses.html" class="external-link">getMarkerClasses</a></span><span class="op">(</span><span class="va">res_control</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span><span class="op">]</span>, <span class="st">"grey"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">mycols</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/pRoloc/man/getMarkerClasses.html" class="external-link">getMarkerClasses</a></span><span class="op">(</span><span class="va">res_control</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, <span class="st">"unknown"</span><span class="op">)</span>

<span class="co">## Create a list of the datasets for plotTranslocations</span>
<span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">res_dl_control</span>, <span class="va">res_dl_treatment</span><span class="op">)</span>

<span class="fu"><a href="../reference/bandle-plots-translocations.html">plotTranslocations</a></span><span class="op">(</span><span class="va">res</span>, fcol <span class="op">=</span> <span class="st">"bandle.allocation.pred"</span>, col <span class="op">=</span> <span class="va">mycols</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 82 features in common</span></code></pre>
<pre><code><span class="co">## ------------------------------------------------</span>
<span class="co">## If length(fcol) == 1 it is assumed that the</span>
<span class="co">## same fcol is to be used for both datasets</span>
<span class="co">## setting fcol = c(bandle.allocation.pred,bandle.allocation.pred)</span>
<span class="co">## ----------------------------------------------</span></code></pre>
<p><img src="v01-getting-started_files/figure-html/plottlres-1.png" width="175"></p>
<p>We can also use the function <code>plotTable</code> to display a summary table of the number of proteins that have changed in location between conditions.</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span><span class="va">tbl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle-plots-translocations-table.html">plotTable</a></span><span class="op">(</span><span class="va">res</span>, fcol <span class="op">=</span> <span class="st">"bandle.allocation.pred"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##        Condition1    Condition2 value</span>
<span class="co">## 1              ER         Golgi     1</span>
<span class="co">## 3              ER       Nucleus     4</span>
<span class="co">## 4              ER    Peroxisome     7</span>
<span class="co">## 5              ER            PM     4</span>
<span class="co">## 6              ER    Proteasome     4</span>
<span class="co">## 7              ER  Ribosome 40S     1</span>
<span class="co">## 9              ER       unknown     1</span>
<span class="co">## 10             ER  Cytoskeleton     1</span>
<span class="co">## 11             ER      Lysosome     6</span>
<span class="co">## 12          Golgi            ER     1</span>
<span class="co">## 13          Golgi mitochondrion     3</span>
<span class="co">## 14          Golgi       Nucleus     2</span>
<span class="co">## 18          Golgi  Ribosome 40S     1</span>
<span class="co">## 19          Golgi  Ribosome 60S     1</span>
<span class="co">## 20          Golgi       unknown     4</span>
<span class="co">## 25  mitochondrion       Nucleus     1</span>
<span class="co">## 31  mitochondrion       unknown     1</span>
<span class="co">## 32  mitochondrion  Cytoskeleton     1</span>
<span class="co">## 37        Nucleus    Peroxisome     1</span>
<span class="co">## 46     Peroxisome         Golgi     1</span>
<span class="co">## 57             PM         Golgi     1</span>
<span class="co">## 61             PM    Proteasome     5</span>
<span class="co">## 64             PM       unknown     1</span>
<span class="co">## 67     Proteasome            ER     1</span>
<span class="co">## 68     Proteasome         Golgi     1</span>
<span class="co">## 70     Proteasome       Nucleus     1</span>
<span class="co">## 72     Proteasome            PM     2</span>
<span class="co">## 73     Proteasome  Ribosome 40S     1</span>
<span class="co">## 74     Proteasome  Ribosome 60S     2</span>
<span class="co">## 79   Ribosome 40S         Golgi     1</span>
<span class="co">## 80   Ribosome 40S mitochondrion     1</span>
<span class="co">## 81   Ribosome 40S       Nucleus     1</span>
<span class="co">## 84   Ribosome 40S    Proteasome     1</span>
<span class="co">## 85   Ribosome 40S  Ribosome 60S     1</span>
<span class="co">## 86   Ribosome 40S       unknown     2</span>
<span class="co">## 95   Ribosome 60S    Proteasome     2</span>
<span class="co">## 100       unknown            ER     1</span>
<span class="co">## 101       unknown         Golgi     2</span>
<span class="co">## 102       unknown mitochondrion     1</span>
<span class="co">## 103       unknown       Nucleus     1</span>
<span class="co">## 105       unknown            PM     1</span>
<span class="co">## 106       unknown    Proteasome     2</span>
<span class="co">## 109       unknown  Cytoskeleton     1</span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="assessing-convergence">Assessing convergence<a class="anchor" aria-label="anchor" href="#assessing-convergence"></a>
</h2>
<p>In this section, we demonstrate how to visually assess convergence of the MCMC algorithm. In the chunk below, we use 4 chains so that we can assess the convergence of the method.</p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">control</span> <span class="op">&lt;-</span> <span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span> 
<span class="va">treatment</span> <span class="op">&lt;-</span> <span class="va">tansim</span><span class="op">$</span><span class="va">lopitrep</span><span class="op">[</span><span class="fl">4</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span>

<span class="va">bandleres</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle.html">bandle</a></span><span class="op">(</span>objectCond1 <span class="op">=</span> <span class="va">control</span>,
                    objectCond2 <span class="op">=</span> <span class="va">treatment</span>,
                    numIter <span class="op">=</span> <span class="fl">50</span>, <span class="co"># usually 10,000</span>
                    burnin <span class="op">=</span> <span class="fl">10L</span>, <span class="co"># usually 5,000</span>
                    thin <span class="op">=</span> <span class="fl">1L</span>, <span class="co"># usually 20</span>
                    gpParams <span class="op">=</span> <span class="va">gpParams</span>,
                    pcPrior <span class="op">=</span> <span class="va">pc_prior</span>,
                    numChains <span class="op">=</span> <span class="fl">4</span>,
                    dirPrior <span class="op">=</span> <span class="va">dirPrior</span><span class="op">)</span></code></pre></div>
<p>We then use the <code>plotConvergence</code> function which plots the ranks of the total number of outliers within in each of the chains. If convergence has been reached these plots should be uniform. As we can see these plots are not uniform and are skewed towards the extremes, so they have not converged. Clearly one of the chains has higher values on average than the other chains producing the skew. Running the algorithm for more iterations (typically 10,000) should produce convergence. This is not done here for brevity of the vignette.</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/graphics/par.html" class="external-link">par</a></span><span class="op">(</span>mfrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>, <span class="fl">2</span><span class="op">)</span><span class="op">)</span>
<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/bandle-plots-convergence.html">plotConvergence</a></span><span class="op">(</span><span class="va">bandleres</span><span class="op">)</span></code></pre></div>
<p><img src="v01-getting-started_files/figure-html/convergence-1.png" width="175"></p>
</div>
<div class="section level2">
<h2 id="description-of-bandle-parameters">Description of <code>bandle</code> parameters<a class="anchor" aria-label="anchor" href="#description-of-bandle-parameters"></a>
</h2>
<p>The <code>bandle</code> function has a significant number of parameters to allow flexible and bespoke analysis. Here, we describe these parameters in more detail to allow user to make decisions on the level of flexibility they wish to exploit.</p>
<ol style="list-style-type: decimal">
<li><p><code>objectCond1</code>. This is a list of <code>MSnSets</code> containing the first condition.</p></li>
<li>
<p><code>objectCond2</code>. This is a list of <code>MSnSets</code> containing the second condition.</p>
<ol style="list-style-type: lower-alpha">
<li>These object should have the same observations and features. These will be checked during bandle analysis.</li>
</ol>
</li>
<li><p><code>fcol</code> indicates the feature column in the <code>MSnSets</code> that indicated the markers. Proteins that are not markers should be labels <code>unknown</code>. The default is <code>markers</code>.</p></li>
<li><p><code>hyperLearn</code> is the algorithm used to learn the hyperparameters of the Gaussian processes. For speed the default is an optimization algorithm called “LBFGS”, however is users want to perform uncertainty quantification on these parameters we can use Markov-chain Monte Carlo (MCMC) methods. This is implemented using the Metropolis-Hastings algorithm. Though this latter methodology provides more information, it is much more costly. The analysis is expected to take several days rather than hours.</p></li>
<li><p><code>numIter</code> is the number of MCMC iterations for the algorithm. We typically suggest around 10,000 iterations is plenty for convergence. Though some cases may take longer. If resources are constrained, we suggest 4,000 iterations as acceptable. A minimum number of iterations is around 1,000 though at this level we expect the posterior estimates to suffer considerably. If possible more parallel chains should be run in this case by changing <code>numChains</code> to, say, 9. The more chains and iterations the more computationally expensive the algorithm. The time taken for the algorithm scales roughly linearly in the number of iterations</p></li>
<li><p><code>burnin</code> is the number of samples that should be discarded from the beginning of the chain due to the bias induced by the starting point of the algorithm. We suggest sensible <code>burnin</code> values to be roughly <span class="math inline">\(10-50\%\)</span> of the number of iterations</p></li>
<li><p><code>thin</code> reduces auto-correlation in the MCMC samples. The default is <span class="math inline">\(5\)</span>, which means every 5th sample is taken. If memory requirements are an issue, we suggest to increase the thinning amount. Though above <span class="math inline">\(20\)</span>, you will see a decrease in performance.</p></li>
<li><p><code>u</code> and <code>v</code> represent the prior hyperparameters of the proportion of outliers. This is modelled using a <code>Beta(u,v)</code> with <code>u = 2</code> and <code>v = 10</code> a default. This suggest that roughly <span class="math inline">\(\frac{u}{u = V} = 16%\)</span> of proteins are believed to be outliers and that it is quite unlikely that more than <span class="math inline">\(50%\)</span> of proteins are outliers. Users can examine the quantiles of the <code>Beta(u,v)</code> distribution if they wish to place a more bespoke prior. For example, increasing <code>u</code> will increase the number of a prior believed outliers.</p></li>
<li><p><code>lambda</code> is a ridge parameter used for numerical stability and is set to <span class="math inline">\(0.01\)</span>. If you experience the algorithm fails due to numerical issue then you can set this value larger. If you require values above <span class="math inline">\(1\)</span> it is likely that there are other issues with the analysis. We suggest checking the method is appropriate for your problem and opening issue detailing the problems.</p></li>
<li><p><code>gpParams</code> results from fitting Gaussian proccess (Gaussian random fields). We refer the users to those functions. The default is <code>NULL</code> which will fit GPs internally but we recommend setting these outside the bandle function because it leads to more stable results.</p></li>
<li><p><code>hyperIter</code> if the hyperparameters of the GP are learnt using MH algorithm then this is the frequency at which these are updated relative to the bandle algorithm. By default this is unused, but if <code>hyperLearn</code> is set to <code>MH</code> then this proceed at every 20 iterations.</p></li>
<li><p><code>hyperMean</code> is the mean of the log normal prior used on the hyperparameters. Though by default this is not used unless <code>PC</code> is set to false</p></li>
<li><p><code>hyperSd</code> is the standard deviation of the log normal prior used on the hyperparameters. The default is <code>c(1,1,1)</code> for the 3 hyperparameters, increasing these values increases the uncertainty in the prior values of the hyperparameters.</p></li>
<li><p><code>seed</code> is the random-number seed.</p></li>
<li><p><code>pg</code> indicates whether or not to use the Polya-Gamma (PG) prior. The default is false and a Dirichlet prior is used instead. If set to true the <code>pg</code> is used. In which case a default PG prior is used. This prior attempts to match the default Dirichlet prior that is used when PG prior is set to false. The PG prior is more computationally expensive but can provide prior information on correlations</p></li>
<li><p><code>pgPrior</code> is by default NULL. We suggest using the <code>pg_prior</code> function to help set this parameter and the documentation therein. This function uses an empirical approach to compute a sensible default.</p></li>
<li><p><code>tau</code> is a parameter used by the Polya-Gamma prior and we refer to BANDLE manuscript for details. By default it is only used if <code>pg</code> prior is true, when the default becomes <code>0.2</code>. At this value the <code>pg</code> prior is similar to the Dirichlet prior but with information on correlations.</p></li>
<li><p><code>dirPrior</code> is the Dirichlet matrix prior on the correlations. This should be provided as a K by K matrix, where K is the number of subcellular niches. The diagonal component should represent the prior belief that organelles do not re-localise (same compartment), where as the off-diagonal terms represent the prior terms of re-localisation. The <code>prior_pred_dir</code> can be used to provide a prior predictive check based on the provided prior. It is recommended that the off-diagonal terms are at least two orders of magnitude smaller than the diagonal terms. An example is given in the vignette.</p></li>
<li><p><code>maternCov</code> is this true the covariance function is the matern covariance, otherwise a Gaussian covariance is used.</p></li>
<li><p><code>PC</code> indicates whether a penalised complexity (PC) is used. The default is true and otherwise log normal priors are used.</p></li>
<li><p><code>pcPrior</code> is a numeric of length 3 indicating the parameters of the PC prior. The prior is placed on the parameters of length-scale, amplitude, and variance in that order. The default values are <span class="math inline">\(0.5,3,100\)</span>, and increasing the value increases the shrinkage towards straight-lines with zero variance.</p></li>
<li><p><code>nu</code> which defaults to 2 is the smoothness of the matern covariance. By increasing <code>nu</code> you encourage smoother solutions. <code>nu</code> should be an integer, though for values of <code>nu</code> above 3, we have observed numerical instability.</p></li>
<li><p><code>propSd</code> is the standard deviation of the random-walk update used in the MH algorithm. We do not recommend changing this unless you are familiar with Bayesian analysis. The default is <code>c(0.3,0.1,0.05)</code> for the 3 hyperparameters. Changing these will alter the efficiency of the underlying samplers.</p></li>
<li><p><code>numChains</code> is the number of parrallel chains and defaults to 4. We recommend using as much processing resources as you have and frequently have used 9 in practise.</p></li>
<li><p><code>BPPARAM</code> is the BiocParallel back-end which defaults to <code><a href="https://rdrr.io/pkg/BiocParallel/man/register.html" class="external-link">BiocParallel::bpparam()</a></code>. We refer you to the <em><a href="https://bioconductor.org/packages/3.14/BiocParallel" class="external-link">BiocParallel</a></em> package for details on setting this dependent on your computing system.</p></li>
</ol>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<p>All software and respective versions used to produce this document are listed below.</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.1.2 (2021-11-01)</span>
<span class="co">## Platform: x86_64-apple-darwin17.0 (64-bit)</span>
<span class="co">## Running under: macOS Big Sur 10.16</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib</span>
<span class="co">## LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">## [1] en_GB.UTF-8/en_GB.UTF-8/en_GB.UTF-8/C/en_GB.UTF-8/en_GB.UTF-8</span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] viridis_0.6.2        viridisLite_0.4.0    pheatmap_1.0.12     </span>
<span class="co">##  [4] pRolocdata_1.33.1    ggplot2_3.3.5        bandle_0.99.12      </span>
<span class="co">##  [7] pRoloc_1.34.0        BiocParallel_1.28.3  MLInterfaces_1.74.0 </span>
<span class="co">## [10] cluster_2.1.2        annotate_1.72.0      XML_3.99-0.8        </span>
<span class="co">## [13] AnnotationDbi_1.56.2 IRanges_2.28.0       MSnbase_2.20.4      </span>
<span class="co">## [16] ProtGenerics_1.26.0  mzR_2.28.0           Rcpp_1.0.8.3        </span>
<span class="co">## [19] Biobase_2.54.0       S4Vectors_0.32.3     BiocGenerics_0.40.0 </span>
<span class="co">## [22] BiocStyle_2.22.0    </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] utf8_1.2.2             tidyselect_1.1.1       RSQLite_2.2.9         </span>
<span class="co">##   [4] grid_4.1.2             lpSolve_5.6.15         pROC_1.18.0           </span>
<span class="co">##   [7] munsell_0.5.0          codetools_0.2-18       ragg_1.2.2            </span>
<span class="co">##  [10] preprocessCore_1.56.0  future_1.23.0          withr_2.4.3           </span>
<span class="co">##  [13] colorspace_2.0-3       filelock_1.0.2         highr_0.9             </span>
<span class="co">##  [16] knitr_1.37             ggalluvial_0.12.3      rstudioapi_0.13       </span>
<span class="co">##  [19] mzID_1.32.0            listenv_0.8.0          GenomeInfoDbData_1.2.7</span>
<span class="co">##  [22] farver_2.1.0           bit64_4.0.5            rprojroot_2.0.2       </span>
<span class="co">##  [25] coda_0.19-4            parallelly_1.30.0      vctrs_0.3.8           </span>
<span class="co">##  [28] generics_0.1.2         ipred_0.9-12           xfun_0.29             </span>
<span class="co">##  [31] BiocFileCache_2.2.1    randomForest_4.6-14    R6_2.5.1              </span>
<span class="co">##  [34] doParallel_1.0.16      GenomeInfoDb_1.30.1    clue_0.3-60           </span>
<span class="co">##  [37] MsCoreUtils_1.6.0      bitops_1.0-7           cachem_1.0.6          </span>
<span class="co">##  [40] assertthat_0.2.1       scales_1.1.1           nnet_7.3-17           </span>
<span class="co">##  [43] gtable_0.3.0           affy_1.72.0            globals_0.14.0        </span>
<span class="co">##  [46] timeDate_3043.102      rlang_1.0.0            systemfonts_1.0.4     </span>
<span class="co">##  [49] GlobalOptions_0.1.2    splines_4.1.2          ModelMetrics_1.2.2.2  </span>
<span class="co">##  [52] impute_1.68.0          hexbin_1.28.2          BiocManager_1.30.16   </span>
<span class="co">##  [55] yaml_2.2.2             reshape2_1.4.4         caret_6.0-90          </span>
<span class="co">##  [58] tools_4.1.2            lava_1.6.10            bookdown_0.24         </span>
<span class="co">##  [61] affyio_1.64.0          ellipsis_0.3.2         jquerylib_0.1.4       </span>
<span class="co">##  [64] RColorBrewer_1.1-2     proxy_0.4-26           plyr_1.8.6            </span>
<span class="co">##  [67] progress_1.2.2         zlibbioc_1.40.0        purrr_0.3.4           </span>
<span class="co">##  [70] RCurl_1.98-1.5         prettyunits_1.1.1      rpart_4.1.16          </span>
<span class="co">##  [73] sampling_2.9           LaplacesDemon_16.1.6   ggrepel_0.9.1         </span>
<span class="co">##  [76] fs_1.5.2               magrittr_2.0.2         data.table_1.14.2     </span>
<span class="co">##  [79] circlize_0.4.13        pcaMethods_1.86.0      mvtnorm_1.1-3         </span>
<span class="co">##  [82] hms_1.1.1              evaluate_0.14          xtable_1.8-4          </span>
<span class="co">##  [85] mclust_5.4.9           gridExtra_2.3          shape_1.4.6           </span>
<span class="co">##  [88] compiler_4.1.2         biomaRt_2.50.2         tibble_3.1.6          </span>
<span class="co">##  [91] ncdf4_1.19             crayon_1.4.2           htmltools_0.5.2       </span>
<span class="co">##  [94] segmented_1.4-0        tidyr_1.2.0            lubridate_1.8.0       </span>
<span class="co">##  [97] DBI_1.1.2              dbplyr_2.1.1           MASS_7.3-55           </span>
<span class="co">## [100] rappdirs_0.3.3         Matrix_1.4-0           cli_3.1.1             </span>
<span class="co">## [103] vsn_3.62.0             gdata_2.18.0           parallel_4.1.2        </span>
<span class="co">## [106] gower_0.2.2            pkgconfig_2.0.3        pkgdown_2.0.2         </span>
<span class="co">## [109] recipes_0.1.17         MALDIquant_1.21        xml2_1.3.3            </span>
<span class="co">## [112] foreach_1.5.2          bslib_0.3.1            XVector_0.34.0        </span>
<span class="co">## [115] prodlim_2019.11.13     stringr_1.4.0          digest_0.6.29         </span>
<span class="co">## [118] Biostrings_2.62.0      rmarkdown_2.11         dendextend_1.15.2     </span>
<span class="co">## [121] curl_4.3.2             kernlab_0.9-29         gtools_3.9.2          </span>
<span class="co">## [124] lifecycle_1.0.1        nlme_3.1-155           jsonlite_1.7.3        </span>
<span class="co">## [127] desc_1.4.0             limma_3.50.0           fansi_1.0.2           </span>
<span class="co">## [130] pillar_1.7.0           lattice_0.20-45        lbfgs_1.2.1           </span>
<span class="co">## [133] KEGGREST_1.34.0        fastmap_1.1.0          httr_1.4.2            </span>
<span class="co">## [136] survival_3.2-13        glue_1.6.1             FNN_1.1.3             </span>
<span class="co">## [139] png_0.1-7              iterators_1.0.13       bit_4.0.4             </span>
<span class="co">## [142] class_7.3-20           stringi_1.7.6          sass_0.4.0            </span>
<span class="co">## [145] mixtools_1.2.0         blob_1.2.2             textshaping_0.3.6     </span>
<span class="co">## [148] memoise_2.0.1          dplyr_1.0.7            e1071_1.7-9           </span>
<span class="co">## [151] future.apply_1.8.1</span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-bandle" class="csl-entry">
Crook, Oliver M, Colin TR Davies, Laurent Gatto, Paul DW Kirk, and Kathryn S Lilley. 2021. <span>“Inferring Differential Subcellular Localisation in Comparative Spatial Proteomics Using BANDLE.”</span> <em>bioRxiv</em>. <a href="doi:%20https://doi.org/10.1101/2021.01.04.425239" class="external-link">doi: https://doi.org/10.1101/2021.01.04.425239</a>.
</div>
<div id="ref-pRoloc:2014" class="csl-entry">
Gatto, Laurent, Lisa M. Breckels, Samuel Wieczorek, Thomas Burger, and Kathryn S. Lilley. 2014. <span>“Mass-Spectrometry Based Spatial Proteomics Data Analysis Using pRoloc and pRolocdata.”</span> <em>Bioinformatics</em>.
</div>
<div id="ref-Gentleman:2004" class="csl-entry">
Gentleman, Robert C., Vincent J. Carey, Douglas M. Bates, Ben Bolstad, Marcel Dettling, Sandrine Dudoit, Byron Ellis, et al. 2004. <span>“Bioconductor: Open Software Development for Computational Biology and Bioinformatics.”</span> <em>Genome Biol</em> 5 (10): –80. <a href="https://doi.org/10.1186/gb-2004-5-10-r80" class="external-link">https://doi.org/10.1186/gb-2004-5-10-r80</a>.
</div>
<div id="ref-Gilks:1995" class="csl-entry">
Gilks, Walter R, Sylvia Richardson, and David Spiegelhalter. 1995. <em>Markov Chain Monte Carlo in Practice</em>. CRC press.
</div>
<div id="ref-Rstat" class="csl-entry">
R Development Core Team. 2011. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="http://www.R-project.org/" class="external-link">http://www.R-project.org/</a>.
</div>
<div id="ref-Tan:2009" class="csl-entry">
Tan, Denise JL, Heidi Dvinge, Andrew Christoforou, Paul Bertone, Alfonso Martinez Arias, and Kathryn S Lilley. 2009. <span>“Mapping Organelle Proteins and Protein Complexes in Drosophila Melanogaster.”</span> <em>Journal of Proteome Research</em> 8 (6): 2667–78.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Oliver M. Crook, Lisa Breckels.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
